gulpfile.js

```diff
@@ -258,33 +262,36 @@ gulp.task('publish-packages', (cb) => {
       else {
         const packageJson = require(packageJsonFilePath);
         const packageName = packageJson.name;
-        const localPackageVersion = packageJson.version;
-        if (!localPackageVersion) {
-          console.log(`ERROR: "${packageJsonFilePath}" doesn't have a non-empty version property.`);
-          errorPackages++;
-        }
-        else {
-          let npmPackageVersion;
-          try {
-            const npmViewResult = JSON.parse(execSync(`npm view ${packageName} --json`, { stdio: ['pipe', 'pipe', 'ignore'] }));
-            npmPackageVersion = npmViewResult['dist-tags']['latest'];
-          }
-          catch (error) {
-            // This happens if the package doesn't exist in NPM.
-          }
 
-          if (localPackageVersion === npmPackageVersion) {
-            upToDatePackages++;
+        if (!package || package === packageName || packageName.endsWith(`-${package}`)) {
+          const localPackageVersion = packageJson.version;
+          if (!localPackageVersion) {
+            console.log(`ERROR: "${packageJsonFilePath}" doesn't have a version specified.`);
+            errorPackages++;
           }
           else {
-            console.log(`Publishing package "${packageName}" with version "${localPackageVersion}"...`);
+            let npmPackageVersion;
             try {
-              npmInstall(packageFolderPath);
-              execSync(`npm publish`, { cwd: packageFolderPath });
-              publishedPackages++;
+              const npmViewResult = JSON.parse(execSync(`npm view ${packageName} --json`, { stdio: ['pipe', 'pipe', 'ignore'] }));
```

Comment:
Nice, so this takes care of the "check if an up to date package was published" thing?

Reply:
Yeah. We only attempt to publish if we need to.